packages/web3-utils/src/errors.ts

```diff
@@ -155,3 +154,33 @@ export class InvalidTypeAbiInput extends Web3Error {
 		super(value, 'components found but type is not tuple');
 	}
 }
+
+// TODO Type of value for all errors should be any as
+// any value can be passed when an error occurs
+export class InvalidBlockNumberOrTag extends Web3Error {
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	public constructor(value: any) {
+		super(value, 'invalid block number or tag given');
+	}
+}
+
+export class InvalidHexString8Bytes extends Web3Error {
+	public constructor(value: string) {
+		super(value, 'invalid 8 byte hex string given');
+	}
+}
+
+export class InvalidHexString32Bytes extends Web3Error {
+	public constructor(value: string) {
+		super(value, 'invalid 32 byte hex string given');
+	}
+}
+
+export class InvalidFilter extends Web3Error {
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	public constructor(value: any) {
+		// TODO Discuss this naive approach to logging object
+		// Does not account for non JSON properties
+		super(JSON.stringify(value), 'invalid filter given');
```

Comment:
Another error prone handling of possible object `value`s

Reply:
That is fine. Unless we have a proper and standard for handling of the errors have a consistent string values are fine. Once we reach to `logging` aspect of the library we will see how to deal it.

Reply:
https://github.com/ChainSafe/web3.js/issues/4583